capa/engine.py

```diff
@@ -33,7 +45,7 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def evaluate(self, ctx):
+    def evaluate(self, features: FeatureSet) -> "Result":
```

Comment:
we have a circular dep Statement -> Result -> Statement, so we have to use this silly syntax with quotes to defer the resolution of the type.

typically i'd just leave off the type hint, but this is a key routine, so we really want hints here.